airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -222,12 +236,18 @@ def _report_data(
     ) -> Mapping[str, Any]:
         replication_start_date = max(pendulum.parse(self._replication_start_date), pendulum.now("utc").subtract(days=90))
 
-        return {
+        params = {
             "reportType": self.name,
             "marketplaceIds": [self.marketplace_id],
             "dataStartTime": replication_start_date.strftime(DATE_TIME_FORMAT),
         }
 
+        if is_valid_replication_end_date(self._replication_end_date):
+            params["dataEndTime"] = self._replication_end_date
+            params["dataStartTime"] = self._replication_start_date
```

Comment:
```suggestion
```
Why are you setting the `dataStartTime` again? It's already set above.

Reply:
```
replication_start_date = max(pendulum.parse(self._replication_start_date), pendulum.now("utc").subtract(days=90))
```

With above logic, any start date older than 90 days would be overwritten.

If an user puts a start & end date older than 90 days, we would want to honour that.

Reply:
done